### ISSUE TITLE
Merge with more than a single column for `left_on` possible?

### ISSUE THREAD

**Issue by JSpenced** [lancedb/lance#1211]
@eddyxu @changhiskhan I had asked about adding columns in #767 and implementing this got put on hold, but now I have some questions.

 Is the `left_on` and `right_on` only able to select a single column (it seems so)? We typically have datasets with `inference_date` and `id`, where for each day (`inference_date` there are multiple overlapping `id`s. We could make a column like `date_id` and join those two columns together to be able to do the merge. I am just checking if this feature will ever be possible or how best to design this from our end.

Thanks!

**Comment by wjones127** [comment]
Yes, right now we only take a single column to merge on. So for now you need to have a single unique id for each row to be able to use that.

We could definitely add support for merging on a tuple of columns, though, so we can support composite keys like you have. 👍

**Comment by JSpenced** [comment]
@wjones127 Okay, that would be awesome! Any idea on when that might be implemented?
